### PR TITLE
feat: integrate wallets

### PR DESCRIPTION
**Pull request by namgold**

### CODE REVIEW

File: src/components/SwapForm/SwapActionButton/index.tsx [hunk 1]
```diff
@@ -136,10 +136,10 @@ const SwapActionButton: React.FC<Props> = ({
   const toggleApprovalModal = useToggleModal(ApplicationModal.SWAP_APPROVAL)
 
   const handleApproveClick = () => {
-    if (walletKey && ['METAMASK', 'TRUST_WALLET'].includes(walletKey?.toString())) {
-      approveCallback()
-    } else {
+    if (['COIN98', 'BRAVE', 'COINBASE'].includes(walletKey)) {
```
Comment (viet-nv) [inline src/components/SwapForm/SwapActionButton/index.tsx:139]: how about these other wallets which dont have custom allowance like metamask?  we should show approval modal

File: src/components/TransactionConfirmationModal/index.tsx [hunk 2]
```diff
@@ -260,7 +260,7 @@ export function TransactionErrorContent({
   confirmText?: string
 }) {
   const theme = useTheme()
-  const [showDetail, setShowDetail] = useState<boolean>(true)
+  const [showDetail, setShowDetail] = useState<boolean>(false)
```
Comment (viet-nv) [inline src/components/TransactionConfirmationModal/index.tsx:263]: why do we hide it?

Comment (namgold) [reply src/components/TransactionConfirmationModal/index.tsx:263]: We should extract minimal message for user, then if they want to see more, they can expand the raw message. I have improved the friendlyError so I think we dont need to show the raw message on default anymore.

File: src/components/SwapForm/SwapActionButton/index.tsx (hunk 1 above)
Comment (namgold) [reply src/components/SwapForm/SwapActionButton/index.tsx:139]: new added wallets don't support too

File: src/components/Web3ReactManager/index.tsx [hunk 3]
```diff
@@ -21,12 +21,12 @@ export default function Web3ReactManager({ children }: { children: JSX.Element }
   const dispatch = useDispatch()
   /** On user change network from wallet, update chainId in store, only work on EVM wallet */
   useEffect(() => {
-    if (triedEager && chainIdEVM && chainIdState !== chainIdEVM && active && isEVM) {
+    if (triedEager.current && chainIdEVM && chainIdState !== chainIdEVM && active && isEVM) {
```
Comment (viet-nv) [inline src/components/Web3ReactManager/index.tsx:24]: is triedEager a ref? I think it's boolean value, not ref

Comment (namgold) [reply src/components/Web3ReactManager/index.tsx:24]: changed to ref now

Comment (viet-nv) [reply src/components/Web3ReactManager/index.tsx:24]: oh, you update the hooks

File: src/hooks/web3/useChangeNetwork.ts [hunk 4]
```diff
@@ -131,39 +132,103 @@ export function useChangeNetwork() {
...
+    [library?.provider, failureCallback, fetchKyberswapConfig, successCallback, walletEVM.walletKey, connector],
   )
 
   const changeNetwork = useCallback(
     async (
       desiredChainId: ChainId,
       customSuccessCallback?: () => void,
       customFailureCallback?: (error: Error) => void,
-      waitUtilUpdatedChainId = false,
+      waitUtilUpdatedChainId = false, //todo: force all to true
```
Comment (viet-nv) [inline src/hooks/web3/useChangeNetwork.ts:231]: TODO

Comment (namgold) [reply src/hooks/web3/useChangeNetwork.ts:231]: we might need an additional refactor on this later 😅

File: src/utils/index.ts [hunk 5]
```diff
@@ -422,54 +430,26 @@ export const deleteUnique = <T>(array: T[] | undefined, element: T): T[] => {
...
 
 export const isOverriddenWallet = (wallet: SUPPORTED_WALLET) => {
   const injectedType = detectInjectedType()
   return (
-    (wallet === WALLET_KEYS.COIN98 && injectedType === WALLET_KEYS.METAMASK) ||
-    (wallet === WALLET_KEYS.METAMASK && injectedType === WALLET_KEYS.COIN98) ||
-    (wallet === WALLET_KEYS.BRAVE && injectedType === WALLET_KEYS.COIN98) ||
-    (wallet === WALLET_KEYS.COIN98 && injectedType === WALLET_KEYS.BRAVE) ||
-    (wallet === WALLET_KEYS.COINBASE && injectedType === WALLET_KEYS.COIN98) ||
+    (wallet === 'COIN98' && injectedType === 'METAMASK') ||
```
Comment (nguyenhoaidanh) [inline src/utils/index.ts:446]: why don't you add more to enum, but remove all of them 🐳

File: src/components/Web3ReactManager/index.tsx [hunk 6]
```diff
@@ -21,12 +21,12 @@ export default function Web3ReactManager({ children }: { children: JSX.Element }
...
       dispatch(updateChainId(chainIdEVM))
     }
     // Only run on change network from wallet
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [chainIdEVM, triedEager, active])
+  }, [chainIdEVM, triedEager.current, active])
 
   // on page load, do nothing until we've tried to connect to the injected connector
   if (isEVM && !triedEager) {
```
Comment (nguyenhoaidanh) [inline src/components/Web3ReactManager/index.tsx:32]: !triedEager. current

Comment (namgold) [reply src/components/Web3ReactManager/index.tsx:32]: this is dependency list. You sure?

Comment (nguyenhoaidanh) [reply src/components/Web3ReactManager/index.tsx:32]: I mean this line:
 if (isEVM && !triedEager) {

File: src/utils/index.ts (hunk 5 above)
Comment (namgold) [reply src/utils/index.ts:446]: `wallet` is already enum type. `WALLET_KEYS` is a duplicate enum of the existing one. I was remove one and keep one. These `'COIN98'` here is not simply string type but they are enum type already. Dont worry.
image

File: src/components/Web3ReactManager/index.tsx (hunk 6 above)
Comment (namgold) [reply src/components/Web3ReactManager/index.tsx]: fixed

**Review by nguyenhoaidanh — APPROVED** [review]

**Pull request merged**